File: tests/fixtures/edgecase_repo/ruby/module3.rb
require_relative "./module2"

module Example
  class Base3
    def initialize(name)
      @name = name
    end
  end

  class Service3 < Base3
    def handle(payload)
      validate(payload)
      true
    end

    def validate(payload)
      !payload.nil?
    end
  end
end

def helper_3(value)
  value
end

File: tests/fixtures/edgecase_repo/lua/module3.lua
local mod2 = require("module2")

local Base3 = {}
Base3.__index = Base3

function Base3.new(name)
  local self = setmetatable({}, Base3)
  self.name = name
  return self
end

local Service3 = setmetatable({}, { __index = Base3 })

function Service3:handle(payload)
  self:validate(payload)
  return true
end

function Service3:validate(payload)
  return payload ~= nil
end

local function helper3(value)
  return value
end

return Service3
